dll/win32/shell32/dialogs/dialogs.cpp

```diff
@@ -1063,44 +1070,285 @@ int WINAPI RestartDialogEx(HWND hWndOwner, LPCWSTR lpwstrReason, DWORD uFlags, D
+                              hDlg, NULL, 
+                              shell32_hInstance, NULL);
+
+    LoadStringW(shell32_hInstance, detailID, szBuffer, _countof(szBuffer));
+    LoadStringW(shell32_hInstance, titleID, szBuffer2, _countof(szBuffer2));
+
+    /* Associate the tooltip with the tool. */
+    tool.lpszText = szBuffer;
+    SendMessage(hwndTip, TTM_ADDTOOLW , 0, (LPARAM)&tool);
+    SendMessage(hwndTip, TTM_SETTITLEW, TTI_NONE, (LPARAM)szBuffer2);
+    SendMessage(hwndTip, TTM_SETMAXTIPWIDTH, 0, 250);
+}
+
+static BOOL IsFriendlyUIActive(VOID)
+{
+    DWORD dwType, dwValue, dwSize;
+    HKEY hKey;
+    LONG lRet;
+
+    lRet = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                         L"SYSTEM\\CurrentControlSet\\Control\\Windows",
+                         0,
+                         KEY_QUERY_VALUE,
+                         &hKey);
+    if (lRet != ERROR_SUCCESS)
+        return FALSE;
+
+    /* First check an optional ReactOS specific override, that Windows does not check.
+       We use this to allow users pairing 'Server'-configuration with FriendlyShutdown.
+       Otherwise users would have to change CSDVersion or LogonType (side-effects AppCompat) */
+    dwValue = 0;
+    dwSize = sizeof(dwValue);
+    lRet = RegQueryValueExW(hKey,
+                            L"EnforceFriendlyLogoff",
+                            NULL,
+                            &dwType,
+                            (LPBYTE)&dwValue,
+                            &dwSize);
+
+    if (lRet == ERROR_SUCCESS && dwType == REG_DWORD && dwValue == 0x1)
+    {
+        RegCloseKey(hKey);
+        return TRUE;
+    }
+
+    /* Check product version number */
+    dwValue = 0;
+    dwSize = sizeof(dwValue);
+    lRet = RegQueryValueExW(hKey,
+                            L"CSDVersion",
+                            NULL,
+                            &dwType,
+                            (LPBYTE)&dwValue,
+                            &dwSize);
+    RegCloseKey(hKey);
+
+    if (lRet != ERROR_SUCCESS || dwType != REG_DWORD || dwValue != 0x300)
+    {
+        /* Allow Friendly UI only on Workstation */
+        return FALSE;
+    }
+
+    /* Check LogonType value */
+    lRet = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                         L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+                         0,
+                         KEY_QUERY_VALUE,
+                         &hKey);
+    if (lRet != ERROR_SUCCESS)
+        return FALSE;
+
+    dwValue = 0;
+    dwSize = sizeof(dwValue);
+    lRet = RegQueryValueExW(hKey,
+                            L"LogonType",
+                            NULL,
+                            &dwType,
+                            (LPBYTE)&dwValue,
+                            &dwSize);
+    RegCloseKey(hKey);
+
+    if (lRet != ERROR_SUCCESS || dwType != REG_DWORD)
+        return FALSE;
+
+    return (dwValue != 0);
+}
+
+static void FancyLogoffOnInit(HWND hwnd, PLOGOFF_DLG_CONTEXT pContext)
```

Comment:
Considering this is a dialog box, why not set the font in the resource?

Reply:
@learn-more I want to use this specific font for only one static control.